Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -400,8 +394,17 @@ extension _BSONDecoder {
             defer { self.storage.popContainer() }
             return try UUID(from: self)
         case .binary:
-            let binary = try self.unbox(value, as: Binary.self)
-            return try UUID(from: binary)
+            let binary = try self.unboxBSONValue(value, as: Binary.self)
+            do {
+                return try UUID(from: binary)
+            } catch {
+                throw DecodingError.dataCorrupted(
```

Comment:
Yeah, I'm inclined to agree actually. If a user has a db with a bunch of old (but valid BSON) data in it, they should still be able to connect to it with the driver and decode without a problem. It wouldn't really make sense to not allow them to use a generic collection just because their UUIDs are stored in an old way. 

I think updating this here is within the scope of this ticket as well.

Reply:
Well, I think removing the `logicError` from `init(from: Binary)` is within scope here. Changing the error behavior when reading `Binary` from a document vs. initializing it directly may be different.

Reply:
I think we should bring this up in the team meeting this week. It seems wrong that if a user had a deprecated type in a database, pulled it out using our code, and reinserted it, they will have experienced a forced upconversion without any warning.

Reply:
i believe our resolution today is in line with the code we have in this PR. I opened [SWIFT-356](https://jira.mongodb.org/browse/SWIFT-356) so we can have further discussion about the Symbol and DBPointer behavior.

Reply:
:+1: